src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -245,11 +245,22 @@ int dtrace_waited_probe(ObjectMonitor* monitor, Handle obj, JavaThread* thr) {
   return 0;
 }
 
-static PlatformMutex gInflationLocks[256];
+static constexpr size_t inflation_lock_count() {
+  return 256;
+}
+
+// Static storage for an array of PlatformMutex. Each entry is aligned to `alignas(PlatformMutex)`
+// by property of the storage itself being aligned to that requirement and each entry storage being
+// `sizeof(PlatformMutex)` aligned up to `alignof(PlatformMutex)`.
+alignas(PlatformMutex) static uint8_t _inflation_locks[inflation_lock_count()][align_up(sizeof(PlatformMutex), alignof(PlatformMutex))];
```

Comment:
I'm afraid you lost me here. Why does this have to be so convoluted/obscure?

Reply:
I thought `sizeof(T)` always was a multiple of `alignof(T)`. Is there some case where this is not true?

Reply:
Ah, I had assumed it wasn't required. But a quick glance says it is. So I'll remove the `align_up`. Should be more simple now.